query/security/ldap/src/main/java/org/codice/ddf/admin/ldap/discover/LdapQuery.java

```diff
@@ -34,88 +35,118 @@
+        conn = new LdapConnectionField().useDefaultRequired();
+        creds = new LdapBindUserInfo().useDefaultRequired();
+        maxQueryResults = new IntegerField(MAX_QUERY_FIELD_NAME);
+        queryBase = new LdapDistinguishedName(QUERY_BASE_FIELD_NAME);
+        queryBase.isRequired(true);
         query = new LdapQueryField();
-        utils = new LdapTestingUtils();
+        query.isRequired(true);
         updateArgumentPaths();
+
+        utils = new LdapTestingUtils();
     }
 
     @Override
     public List<DataType> getArguments() {
-        // TODO: tbatie - 4/3/17 - Add other args
-        return ImmutableList.of(conn, creds, dn, maxQueryResults, query);
+        return ImmutableList.of(conn, creds, maxQueryResults, queryBase, query);
     }
 
     @Override
     public ListField<MapField> performFunction() {
-
-        LdapConnectionAttempt connectionAttempt = utils.bindUserToLdapConnection(conn, creds);
-        addResultMessages(connectionAttempt.messages());
-
-        if(!connectionAttempt.connection().isPresent()) {
-            return null;
-        }
-
-        List<SearchResultEntry> searchResults = utils.getLdapQueryResults(
-                connectionAttempt.connection().get(),
-                dn.getValue(),
-                query.getValue(),
-                SearchScope.WHOLE_SUBTREE,
-                maxQueryResults.getValue());
-
+        List<SearchResultEntry> searchResults;
         List<MapField> convertedSearchResults = new ArrayList<>();
+        ListField<MapField> entries;
+        try (LdapConnectionAttempt connectionAttempt = utils.bindUserToLdapConnection(conn,
+                creds)) {
+            addMessages(connectionAttempt);
 
-        for (SearchResultEntry entry : searchResults) {
-            MapField entryMap = new MapField();
-            for (Attribute attri : entry.getAllAttributes()) {
-                entryMap.put("name",
-                        entry.getName()
-                                .toString());
-                if (!attri.getAttributeDescriptionAsString()
-                        .toLowerCase()
-                        .contains("password")) {
-                    List<String> attributeValueList = attri.parallelStream()
-                            .map(ByteString::toString)
-                            .collect(Collectors.toList());
-                    String attributeValue = attributeValueList.size() == 1 ? attributeValueList.get(
-                            0) : attributeValueList.toString();
-                    entryMap.put(attri.getAttributeDescriptionAsString(), attributeValue);
+            if (containsErrorMsgs()) {
+                return null;
+            }
+
+            searchResults = utils.getLdapQueryResults(connectionAttempt.result(),
+                    queryBase.getValue(),
+                    query.getValue(),
+                    SearchScope.WHOLE_SUBTREE,
+                    maxQueryResults.getValue() == null ?
```

Comment:
I'd still have to do a null check on `getValue` to see if the default value should be used.